onnx/defs/reduction/defs.cc

```diff
@@ -1,99 +1,101 @@
 
 ONNX_OPERATOR_SCHEMA(ReduceSumSquare)
     .FillUsing(ReduceDocGenerator("sum square"));
 
-ONNX_OPERATOR_SCHEMA(ReduceMean)
-    .FillUsing(ReduceDocGenerator("mean"));
+ONNX_OPERATOR_SCHEMA(ReduceMean).FillUsing(ReduceDocGenerator("mean"));
 
-ONNX_OPERATOR_SCHEMA(ReduceProd)
-    .FillUsing(ReduceDocGenerator("product"));
+ONNX_OPERATOR_SCHEMA(ReduceProd).FillUsing(ReduceDocGenerator("product"));
 
-ONNX_OPERATOR_SCHEMA(ReduceLogSum)
-    .FillUsing(ReduceDocGenerator("log sum"));
+ONNX_OPERATOR_SCHEMA(ReduceLogSum).FillUsing(ReduceDocGenerator("log sum"));
 
 ONNX_OPERATOR_SCHEMA(ReduceLogSumExp)
     .FillUsing(ReduceDocGenerator("log sum exponent"));
 
-ONNX_OPERATOR_SCHEMA(ReduceL1)
-    .FillUsing(ReduceDocGenerator("L1 norm"));
-
-ONNX_OPERATOR_SCHEMA(ReduceL2)
-    .FillUsing(ReduceDocGenerator("L2 norm"));
+ONNX_OPERATOR_SCHEMA(ReduceL1).FillUsing(ReduceDocGenerator("L1 norm"));
 
+ONNX_OPERATOR_SCHEMA(ReduceL2).FillUsing(ReduceDocGenerator("L2 norm"));
 
-}  // namespace ONNX_NAMESPACE
+} // namespace ONNX_NAMESPACE
 
 namespace ONNX_NAMESPACE {
 
 std::function<void(OpSchema&)> ArgReduceDocGenerator(const char* name) {
-    return [=](OpSchema& schema) {
-        std::string doc = R"DOC(
+  return [=](OpSchema& schema) {
+    std::string doc = R"DOC(
 Computes the indices of the {name} elements of the input tensor's element along the 
 provided axis. The resulted tensor has the same rank as the input if keepdims equal 1.
 If keepdims equal 0, then the resulted tensor have the reduced dimension pruned. 
 The type of the output tensor is integer.)DOC";
-        ReplaceAll(doc, "{name}", name);
-        schema.SetDoc(doc);
-        schema.Attr("axis",
-                    "The axis in which to compute the arg indices",
-                    AttributeProto::INT,
-                    OPTIONAL);
-        schema.Attr("keepdims",
-                    "Keep the reduced dimension or not, default 1 mean keep reduced dimension.",
-                    AttributeProto::INT,
-                    static_cast<int64_t>(1));
-        schema.Input(0, "data", "An input tensor.", "T");
-        schema.Output(0, "reduced", "Reduced output tensor with integer data type.", "tensor(int32)");
-        schema.TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" }, "Constrain input and output types to float tensors.");
-    };
+    ReplaceAll(doc, "{name}", name);
+    schema.SetDoc(doc);
+    schema.Attr(
+        "axis",
+        "The axis in which to compute the arg indices",
+        AttributeProto::INT,
+        static_cast<int64_t>(0));
+    schema.Attr(
+        "keepdims",
+        "Keep the reduced dimension or not, default 1 mean keep reduced dimension.",
+        AttributeProto::INT,
+        static_cast<int64_t>(1));
+    schema.Input(0, "data", "An input tensor.", "T");
+    schema.Output(
+        0,
+        "reduced",
+        "Reduced output tensor with integer data type.",
+        "tensor(uint64)");
```

Comment:
shall we use int64 here to be consistent with topk?

Reply:
I think, we probably should change topK's indices (2nd output) with type "uint64" too, since there's no negative index. :)

Reply:
I had some offline discussions with @dzhulgakov, we think it's better to use int64 to represent indices:

1. some system doesn't support uint64 (e.g. Java)
2. in case error happened, say -1 is returned, it's easier to discover when it's -1 instead of wraparound

Reply:
Another data point is from tensorflow, they also only support signed int for the indices.

For example, in their argmax/argmin, the return values (indices) are either int or int64. I think signed int is more expressive when we need to return some special values/errors.